### ISSUE TITLE
Replace use of std::regex with google re2

### ISSUE THREAD

**Issue by ceztko** [podofo/podofo#46]
`std::regex` is used [here](https://github.com/podofo/podofo/blob/808a995be84197e9d29b2fdf6e55422ba5828846/src/podofo/private/PdfDeclarationsPrivate.cpp#L516) and [here](https://github.com/podofo/podofo/blob/808a995be84197e9d29b2fdf6e55422ba5828846/src/podofo/main/PdfPage_TextExtraction.cpp#L556). We could migrate to a faster solution that has some unicode support, such as google [re2](https://github.com/google/re2), at the cost of loosing "full" ECMAScript support (really? is it really that useful or most notations we used are supported by google re2 as well?). Consider adding it as a submodule. Related to #30

**Comment by derreisende77** [comment]
As I wrote in another issue you should as well consider [compile time regular expressions](https://github.com/hanickadot/compile-time-regular-expressions). I found [this paper](https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2019/p1433r0.pdf) which contains some benchmark data compared to re2 - where performance seems to be equal or just a little worse. However, CTRE is available as 3 .hpp files and does support unicode vs a whole bunch of files for re2.

**Comment by ceztko** [comment]
Ctre does not do dynamic regex, and we need those in rext extraction. I am unsure we should carry a second dependency just to support a single static re. If there were more, I could change my mind, but not now.

**Comment by ceztko** [comment]
To provide more context, I [asked](https://github.com/hanickadot/compile-time-regular-expressions/issues/276) ctre dev if it supports ECMAScript and if it supports dynamic patterns. She said no to both questions and she recommended re2 instead.

**Comment by derreisende77** [comment]
Yeah it might have been beneficial but if re2 is better suited why not.

**Comment by davidbolvansky** [comment]
consider pcre2 over re2
